lib/ts/recipeImplementation.ts

```diff
@@ -137,6 +145,62 @@ export default function RecipeImplementation(recipeImplInput: {
             }
 
             // we do not send an event here since it's triggered in setIdRefreshToken area.
+        },
+
+        getInvalidClaimsFromResponse: async function(input: {
+            response: AxiosResponse | Response;
+        }): Promise<ClaimValidationError[]> {
+            let body;
+            if ("body" in input.response) {
+                body = await input.response.clone().json();
+            } else {
+                body = input.response.data;
+            }
+
+            return body.claimValidationErrors;
+        },
+
+        getGlobalClaimValidators: async function(input: {
+            claimValidatorsAddedByOtherRecipes: SessionClaimValidator[];
+        }) {
+            return input.claimValidatorsAddedByOtherRecipes;
+        },
+
+        validateClaims: async function(input: {
+            claimValidators: SessionClaimValidator[];
```

Comment:
shouldn't this be similar to how it's in node? Instead of taking an array here, we should take an optional function like `overrideGlobalClaimValidators`?

Reply:
We should also take a lock while validating (as discussed)